### PR TITLE
Sequence for Connection IDs, RETIRE

### PR DESCRIPTION
**Pull request by MikeBishop**
Suggestions for your PR, as discussed.

### CODE REVIEW

File: draft-ietf-quic-transport.md [hunk 1]
```diff
@@ -1249,8 +1252,11 @@ using the NEW_CONNECTION_ID frame ({{frame-new-connection-id}}).
 ### Issuing Connection IDs
 
 The initial connection ID issued by an endpoint is the Source Connection ID
-during the handshake.  Subsequent connection IDs are communicated to the peer
-using NEW_CONNECTION_ID frames ({{frame-new-connection-id}}).
+during the handshake.  The sequence number of the initial connection ID is 0. If
```
[the remark this answers is not in the document]
Comment (mikkelfj) [reply draft-ietf-quic-transport.md]: That would result in complications with non-unique 5-tuples if I understand you correctly.

[the remark this answers is not in the document]
Comment (ianswett) [reply draft-ietf-quic-transport.md]: I was thinking that the handshake connection IDs exist, but they don't need an identifier?

[the remark this answers is not in the document]
Comment (mikkelfj) [reply draft-ietf-quic-transport.md]: Ah, the id of the id, that is different.

[the remark this answers is not in the document]
Comment (MikeBishop) [reply draft-ietf-quic-transport.md]: We can avoid that if the entire CID is sent in the RETIRE_CID frame.  That would mean, instead, that issuers need to stop using the contained CID for the current connection iff they are and absorb duplicates by silently consuming RETIRE messages for CIDs which aren't associated with the connection.

By having the sequence number in the RETIRE_CID frame, we make it shorter (one varint) and enable using the same deduplicating code as the consumer of NEW_CID messages.

[the remark this answers is not in the document]
Comment (janaiyengar) [reply draft-ietf-quic-transport.md]: I like the use of seq num here. It does serve the dedup purpose, and is easier on implementations as well.

File: draft-ietf-quic-transport.md [hunk 2]
```diff
@@ -1249,8 +1252,11 @@ using the NEW_CONNECTION_ID frame ({{frame-new-connection-id}}).
...
+the preferred_address transport parameter is sent, the sequence number of the
+supplied connection ID is 1. Subsequent connection IDs are communicated to the
+peer using NEW_CONNECTION_ID frames ({{frame-new-connection-id}}), and the
+sequence number on each newly-issued connection ID MUST increase by 1.
```
Comment (janaiyengar) [inline draft-ietf-quic-transport.md]: I think this should say something about CIDs on Retry (that they don't really count).

**Pull request merged**